### PR TITLE
updated to allow more than one gateway config

### PR DESCRIPTION
**Pull request by darren221980**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSPO-1355

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -8,7 +8,7 @@ data "local_file" "configuration" {
 
 data "azurerm_key_vault" "certificate_vault" {
   name                = var.vault_name
-  resource_group_name = var.key_vault_resource_group
+  resource_group_name = var.env == "perftest" || var.env == "aat" ? "core-infra-${var.subscription}-rg" : "core-infra-${var.env}-rg"
```
Comment (msl8r) [inline main.tf:11]: This won't work for Shared services as we are using ss-dev-network-rg, we pass the resource group to the module using key_vault_resource_group

File: config.tf [hunk 2]
```diff
@@ -1,14 +1,50 @@
...
+    "solutionOwner"      = "CFT"
+    "activityName"       = "AKS"
+    "dataClassification" = "Internal"
+    "automation"         = ""
+    "costCentre"         = "10245117" // until we get a better one, this is the generic cft contingency one
+    "environment"        = local.env_display_names[var.env]
+    "criticality"        = local.criticality[var.env]
+  }
+}
+
```
Comment (msl8r) [inline config.tf:36]: This was moved out as it will be different for each environment

File: secret.tf [hunk 3]
```diff
@@ -1,12 +1,13 @@
 data "azurerm_key_vault" "main" {
   name                = var.vault_name
-  resource_group_name = var.key_vault_resource_group
+  resource_group_name = "core-infra-${var.subscription}-rg"
```
Comment (msl8r) [inline secret.tf:3]: This should stay as a variable

File: config.tf (hunk 2 above)
Comment (thomast1906) [reply config.tf]: Moved into CFT component

File: secret.tf (hunk 3 above)
Comment (thomast1906) [reply secret.tf]: Updated to variable to match module

**Review by zc-hmcts — APPROVED** [review]
changes have been tested successfully against shared services in sandbox and dev

**Pull request merged**